flink-runtime/src/main/java/org/apache/flink/runtime/shuffle/ShuffleEnvironment.java

```diff
@@ -163,4 +167,48 @@
 	boolean updatePartitionInfo(
 		ExecutionAttemptID consumerID,
 		PartitionInfo partitionInfo) throws IOException, InterruptedException;
+
+	/**
+	 * Task executor local context of shuffle service used to create {@link ShuffleEnvironment}.
+	 */
+	class ShuffleLocalContext {
```

Comment:
This class seems not be used atm. What is the motivation for bringing this?

Reply:
true, some rebase leftover, it is now `ShuffleEnvironmentContext`